inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -80,9 +88,94 @@ void MKLDNNPadNode::initSupportedPrimitiveDescriptors() {
     config.outConfs[0].constant = false;
 
     auto memoryFormat = MKLDNNMemory::GetPlainFormat(getParentEdgeAt(0)->getDims());
-    config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), dataType, memoryFormat);
-    config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), dataType, memoryFormat);
-    supportedPrimitiveDescriptors.push_back({config, impl_desc_type::unknown, memoryFormat});
+    auto pushSupportedPrimitiveDescriptor = [&](memory::format memoryFormat) {
+        config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), dataType, memoryFormat);
+        config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), dataType, memoryFormat);
+        supportedPrimitiveDescriptors.push_back({config, impl_desc_type::ref, memoryFormat});
+    };
+    pushSupportedPrimitiveDescriptor(memoryFormat);
+
+    if (numOfDims != 4 && numOfDims != 5)
+        return;
+
+    if (numOfDims == 4) {
+        pushSupportedPrimitiveDescriptor(mkldnn::memory::nhwc);
+
+        if (srcDims[1] % 8 == 0)
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::nChw8c);
+        if (srcDims[1] % 16 == 0)
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::nChw16c);
+    } else if (numOfDims == 5) {
+        pushSupportedPrimitiveDescriptor(mkldnn::memory::ndhwc);
+
+        if (srcDims[1] % 8 == 0)
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::nCdhw8c);
+        if (srcDims[1] % 16 == 0)
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::nCdhw16c);
+    }
+}
+
+void MKLDNNPadNode::selectOptimalPrimitiveDescriptor() {
+    if (supportedPrimitiveDescriptors.size() == 1) {
+        selectPrimitiveDescriptorByIndex(0);
+        return;
+    }
+
+    mkldnn::memory::format memoryFormat = MKLDNNMemory::GetPlainFormat(getChildEdgeAt(0)->getDims());
+    auto parent = getParentEdgeAt(0)->getParent();
+    if (parent->getSelectedPrimitiveDescriptor() != nullptr) {
+        int outputIndex = getParentEdgeAt(0)->getOutputNum();
+        if (outputIndex < 0)
+            THROW_IE_EXCEPTION << "Cannot find index of output node";
```

Comment:
BTW, the same comment as for another review: https://github.com/openvinotoolkit/openvino/pull/3449#discussion_r539979404